Visual Studio Code (1.98.0, undefined, desktop)
Jupyter Extension Version: 2025.2.0.
Python Extension Version: 2025.2.0.
Pylance Extension Version: 2025.3.1.
Platform: linux (x64).
Home = /home/arnold
Temp Storage folder ~/.config/Code/User/globalStorage/ms-toolsai.jupyter/version-2025.2.0
Workspace folder ~/wd
00:46:24.839 [info] Starting Kernel (Python Path: ~/.virtualenvs/virt1/bin/python, VirtualEnvWrapper, 3.13.2) for '~/wd/programming/python/~LearnPython/practice_ground.ipynb' (disableUI=true)
00:46:26.331 [info] Process Execution: ~/.virtualenvs/virt1/bin/python -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
00:46:26.377 [info] Process Execution: ~/.virtualenvs/virt1/bin/python -m ipykernel_launcher --f=/run/user/1000/jupyter/runtime/kernel-v38550a45a67778d17614e4af8eb73ced16fa56ed5.json
    > cwd: ~/wd/programming/python/~LearnPython
00:46:26.472 [info] Process Execution: ~/.virtualenvs/virt1/bin/python -m pip list
00:46:30.082 [info] Kernel successfully started
00:46:30.114 [info] Process Execution: ~/.virtualenvs/virt1/bin/python /home/~/.vscode/extensions/ms-toolsai.jupyter-2025.2.0-linux-x64/pythonFiles/printJupyterDataDir.py
00:49:44.882 [warn] Cell completed with errors [su [Error]: name 'np' is not defined
	at n.execute (/home/~/.vscode/extensions/ms-toolsai.jupyter-2025.2.0-linux-x64/dist/extension.node.js:294:4958)] {
  ename: 'NameError',
  evalue: "name 'np' is not defined",
  traceback: [
    '\x1B[31m---------------------------------------------------------------------------\x1B[39m',
    '\x1B[31mNameError\x1B[39m                                 Traceback (most recent call last)',
    '\x1B[36mCell\x1B[39m\x1B[36m \x1B[39m\x1B[32mIn[1]\x1B[39m\x1B[32m, line 2\x1B[39m\n' +
      '\x1B[32m      1\x1B[39m \x1B[38;5;66;03m# One-dimensional subarrays\x1B[39;00m\n' +
      '\x1B[32m----> \x1B[39m\x1B[32m2\x1B[39m x = \x1B[43mnp\x1B[49m.arange(\x1B[32m10\x1B[39m)\n' +
      '\x1B[32m      3\x1B[39m \x1B[38;5;28mprint\x1B[39m(x)\n',
    "\x1B[31mNameError\x1B[39m: name 'np' is not defined"
  ]
}
00:49:51.012 [warn] Cell completed with errors [su [Error]: division by zero
	at n.execute (/home/~/.vscode/extensions/ms-toolsai.jupyter-2025.2.0-linux-x64/dist/extension.node.js:294:4958)] {
  ename: 'ZeroDivisionError',
  evalue: 'division by zero',
  traceback: [
    'Traceback \x1B[36m(most recent call last)\x1B[39m:\n',
    '  Cell \x1B[92mIn[7]\x1B[39m\x1B[92m, line 1\x1B[39m\n    func2(1)\n',
    '  Cell \x1B[92mIn[5]\x1B[39m\x1B[92m, line 6\x1B[39m in \x1B[95mfunc2\x1B[39m\n' +
      '    return func1(a, b)\n',
    '\x1B[36m  \x1B[39m\x1B[36mCell\x1B[39m\x1B[36m \x1B[39m\x1B[32mIn[5]\x1B[39m\x1B[32m, line 2\x1B[39m\x1B[36m in \x1B[39m\x1B[35mfunc1\x1B[39m\n' +
      '\x1B[31m    \x1B[39m\x1B[31mreturn a / b\x1B[39m\n',
    '\x1B[31mZeroDivisionError\x1B[39m\x1B[31m:\x1B[39m division by zero\n'
  ]
}
00:49:51.025 [warn] Cancel all remaining cells due to cancellation or failure in execution
